socket.go

```diff
@@ -336,6 +337,7 @@ const (
 	SendTimeout
 	QueryAcceptConnections
 	BindToDevice
+	TcpNoDelay
```

Comment:
WasmEdge has been adding socket options to this list and incrementing the value as they go: https://github.com/WasmEdge/WasmEdge/blob/d649efad4a8d33e5302fdc24c6e8e823b08e9369/thirdparty/wasi/api.hpp#L1723 (most recently in https://github.com/WasmEdge/WasmEdge/commit/095136362a2f88a076bee83f05c96207038462ea). There's a good chance that they'll add a socket option that clashes with this.

If we're diverging from WasmEdge, let's give the options a high value:

```go
const (
	TcpNoDelay SocketOption = 0x10000 + iota
)
```

If WasmEdge adds `TCP_NODELAY` and it has a different value then we can handle that in the host module.

Reply:
Or as @achille-roussel mentioned above, better yet we submit PRs to keep the two in sync.

Reply:
yeah that makes sense. Will open an issue and PR on the wasmedge side and see what they think/want.